src/helpers/map.ts

```diff
@@ -0,0 +1,34 @@
+export function mapUndefinedPipe<T, U1>(
+  source: T | undefined,
+  fn1: (_: NonNullable<T>) => U1 | undefined
+): U1 | undefined;
+export function mapUndefinedPipe<T, U1, U2>(
+  source: T | undefined,
+  fn1: (_: NonNullable<T>) => U1 | undefined,
+  fn2: (_: NonNullable<U1>) => U2 | undefined
+): U2 | undefined;
+export function mapUndefinedPipe<T, U1, U2, U3>(
+  source: T | undefined,
+  fn1: (_: NonNullable<T>) => U1 | undefined,
+  fn2: (_: NonNullable<U1>) => U2 | undefined,
+  fn3: (_: NonNullable<U2>) => U3 | undefined
+): U3 | undefined;
+// eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/no-explicit-any
+export function mapUndefinedPipe(source: any, ...fn: Function[]) {
```

Comment:
パイプの概念やmapがMapなのかarray.mapなのかややこしいかもなので、この関数がやっていることの説明をちょっとだけ書いておくのどうでしょう。
```suggestion
/**
 * 一連の関数を実効する。途中でundefinedを返すとその後undefinedを返す。
 */
// eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/no-explicit-any
export function mapUndefinedPipe(source: any, ...fn: Function[]) {
```